rule/merge.go

```diff
@@ -408,18 +405,10 @@ func squashDict(x, y *bzl.DictExpr) (*bzl.DictExpr, error) {
 	}
 
 	keys := make([]string, 0, len(cases))
-	haveDefault := false
 	for k := range cases {
-		if k == "//conditions:default" {
-			haveDefault = true
-			continue
-		}
 		keys = append(keys, k)
 	}
 	sort.Strings(keys)
-	if haveDefault {
-		keys = append(keys, "//conditions:default") // must be last
```

Comment:
This comments basically tells me this change is wrong. But it is not clear why it **must** be last.

Reply:
Documentation for `select` is here https://docs.bazel.build/versions/master/be/functions.html#select

Reply:
`bazel test //...` passes. So this "must" does not have a test 🤷‍♂️

Reply:
"should" would be a more precise word. There's no technical reason `"//conditions:default"` needs to be at the bottom, but it's where it usually goes.